### PR TITLE
[JENKINS-62515] - Introduce CSS variables in the Login screen CSS

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This patch introduces CSS variables in the Login Page CSS, similar to what #4752 did for the main CSS files. It implies that the colors will not be picked up on Internet Explorer 11, but it still retains good look and feel.

Chrome, with Dark Theme:

image

Internet Explorer 11:

image

### Proposed changelog entries

* Add support for Dark Theme in the login screen.
** https://github.com/jenkinsci/dark-theme/

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: war/src/main/webapp/css/simple-page.theme.css [hunk 1]
```diff
@@ -42,25 +38,29 @@ html {
   height: 130px;
 }
 
-.simple-page form .submit input:focus {
-  background-color: #2171ce; /*var(--main-bg-color-focus);*/
-  border: solid 1px #2171ce; /*var(--main-bg-color-focus);*/
+.simple-page form .submit input:hover {
```
Comment (oleg-nenashev) [inline war/src/main/webapp/css/simple-page.theme.css:41]: Unrelated update. Before this change the button was not hightlighted on hover like Jenkins form buttons. Maybe we need both Hover and Focus for accessibility, not sure

**Review by fqueiruga — CHANGES_REQUESTED** [review]
In my opinion the degradation on IE11 does not cut it. There should be fallbacks, either done manually or by moving this CSS to webpack.

File: war/src/main/webapp/css/simple-page.theme.css [hunk 2]
```diff
@@ -42,25 +38,29 @@ html {
...
+  background-color: var(--btn-primary-bg-focus);
```
Comment (fqueiruga) [inline war/src/main/webapp/css/simple-page.theme.css:42]: Focus should be the same as hover but adding an outline. Basically there's never going to be a focus variable modifier, just hover and active. Variable names should be consistent within Jenkins core.

**Review by fqueiruga — CHANGES_REQUESTED** [review]

**Comment by timja** [comment]
> In my opinion the degradation on IE11 does not cut it. There should be fallbacks, either done manually or by moving this CSS to webpack.

👍 I wasn't sure if that was what it normally looked like as the image looks okay, moving to less would be nice, otherwise manual fallback is an easy fix

**Comment by fqueiruga** [comment]
Also I want to question whether it is appropiate to have a dark theme for the login screen. Isn't dark theme a user preference anyway?

File: war/src/main/webapp/css/simple-page-variables.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+:root {
+    --background: #ffffff;
+    --background-medium: (--background);
```
Comment (timja) [inline war/src/main/webapp/css/simple-page-variables.css:3]:
```suggestion
    --background-medium: var(--background);
```

**Comment by timja** [comment]
Created https://github.com/oleg-nenashev/jenkins/pull/50

**Comment by oleg-nenashev** [comment]
Thanks @timja !

**Review by timja — APPROVED** [review]

**Review by fqueiruga — APPROVED** [review]

**Comment by fqueiruga** [comment]
I haven't tried these locally but the code looks good and all concerns have been addressed.

Two things though:
- Can you update the screenshots?
- We should add a ticket on the JIRA epic to port this code to webpack / less

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information

**Review by res0nance — APPROVED** [review]

File: war/src/main/webapp/css/simple-page-forms.css [hunk 4]
```diff
@@ -31,12 +35,17 @@
 }
 .Checkbox .svg-icon {
   opacity: 0;
-  fill: #fff;
```
Comment (Wadeck) [inline war/src/main/webapp/css/simple-page-forms.css:34]: @oleg-nenashev @timja @fqueiruga 

Was there any reason to remove this "fill" ?

In the login page, the "Keep me signed in" checkbox is less constrasted now.

### Version 2.222

Screenshot_2021-05-27_110854_001

### Version 2.277

Screenshot_2021-05-27_110837_001

Adjustment proposed in https://github.com/jenkinsci/jenkins/pull/5536

Comment (timja) [reply war/src/main/webapp/css/simple-page-forms.css:34]: no, accidental

**Pull request merged**